biogrid/libs/biogrid-simulator/src/lib/building/building.ts

```diff
@@ -23,21 +31,22 @@ export class Building implements EnergyUser {
   /** Defines the resistance of the building due to the wiring */
   gridItemResistance = RESISTANCE.BUILDING;
 
+  private readonly minCapacity: Energy;
+  private readonly maxCapacity: Energy;
+
   /**
    * @param {number} energy Amount of energy the building will have in joules.
    */
-  constructor(energy: number, x: Distance, y: Distance,
-      gridItemName: string = GRID_ITEM_NAMES.ENERGY_USER,
-      private readonly minCapacity: Energy = BUILDING.MIN_CAPACITY,
-      private readonly maxCapacity:Energy = BUILDING.MAX_CAPACITY,
-  ) {
-    this.relativePosition = { x, y };
-    this.gridItemName = gridItemName;
-    if (this.isPositive(energy)) {
-      this.energyInJoules = energy;
+  constructor(buildingParams: BuildingParams) {
+    this.relativePosition = { x: buildingParams.x, y: buildingParams.y };
+    this.gridItemName = buildingParams.gridItemName;
+    if (this.isPositive(buildingParams.energy)) {
+      this.energyInJoules = buildingParams.energy;
     } else {
       throw new Error("Can't create a building with negative energy!");
     }
+    this.minCapacity = buildingParams.minCapacity ? buildingParams.minCapacity : BUILDING.MIN_CAPACITY;
+    this.maxCapacity = buildingParams.maxCapacity ? buildingParams.maxCapacity : BUILDING.MAX_CAPACITY;
```

Comment:
To match the pattern you're using in other files, you can default these using `private readonly minCapacity: Energy = BUILDING.MIN_CAPACITY` above the constructor, and conditionally override the value in the constructor